### ISSUE TITLE
Pipeline vanish after upgrade to 1.35.1

### ISSUE THREAD

**Issue by devoxel** [spinnaker/spinnaker#6970]
### Issue Summary:

When upgrading from Spinnaker 1.34.3 to 1.35.1, we hit an issue where all application pages displayed "No pipelines configured for this application.". 

### Cloud Provider(s):

GCP.

### Environment:

- Self hosted spinnaker on GKE via a Helm chart.
- GCS backed metadata storage for front50.
- We set `pipelineCache.filterFront50Pipelines` & `front50.useTriggeredByEndpoint` to true in our 1.34.3 config (via `-local.yml`)
- We have two spinnaker instances, let's call them `test` and `prod`, which are completely isolated in different GKE projects. For the purposes of debugging, `test` is still running 1.35.1 but `prod` has been reverted to 1.34.3.

I can provide more detail about the environment if required.

### Feature Area:

- Pipelines (particularly front50 storage/retrieval)

### Description:

- Yesterday I upgraded both spinnaker instances we have from Spinnaker 1.34.3 to 1.35.1.
- After we hit an issue where all application pages displayed "No pipelines configured for this application.".
- Using cURL, we can confirm that the gate `/applications/{application}/pipelineConfigs/` API returns an empty list.
- Restarting front50 makes no difference, nor does force refreshing cache.
- We rolled back on the `prod` cluster and the issue went away.

### Additional Details:

- We have some custom tooling to automate spinnaker pipeline creation (spincreator). We ran this on the `test` cluster and this worked, the pipelines were created. This runs periodically on the 1.34 cluster also. We can't solely rely on this tool, since it's just a subset, but it gives a good basis to compare.

- Looking at the difference between the data on GCS for the two different clusters, I noticed that on the 1.35.1 cluster, the pipelines that were recreated by spincreator now have a `pipeline-metadata.json`.

- We only have `specification.json` files for pipelines that weren't touched by spincreator, and in the 1.34.3 instance, we only have specification.json files. Here I illustrate using rough commands I used to learn this.

```
# ( note: this is fish shell, not bash )
# download pipeline data
> gsutil -m cp -r 'gs://test-spinnaker-instance/front50/pipelines/' .
# these corresponds to working pipelines recreated by spincreator
> for l in pipelines/*/pipeline-metadata.json; echo $l; end | wc
    296     296   20720
# these include all pipelines 
> for l in pipelines/*/specification.json; echo $l; end | wc
    397     397   26202
    
> gsutil -m cp -r 'gs://prod-spinnaker-instance/front50/pipelines/' .
# these corresponds to working pipelines recreated by spincreator
> for l in pipelines/*/pipeline-metadata.json; echo $l; end | wc
    0     0   0
# these include all pipelines 
> for l in pipelines/*/specification.json; echo $l; end | wc
   1559    1559  102894
```

- I enabled debug logs and inspected output. There does not seem to be anything interesting here, but perhaps you can point me in the right direction. Due to how much confidential information is in the logs, I can't just dump them all out.

- We have metric collection enabled, here is the output of storageServiceSupport_cacheSize which shows that the number of pipelines & service_account objects drops to 0 after the upgrade. Then we see it go to 296 after our spincreator runs, recreating missing pipelines.

![image](https://github.com/user-attachments/assets/375428a2-cb3a-4ede-9d3f-db53b41f1997)

### Steps to Reproduce:

I have not attempted to create a replication from scratch yet.

I could attempt to reproduce by creating a GCS backed spinnaker at 1.34.3, creating pipelines, and then upgrading.

If this is required to fix the bug, I can take a look.

**Comment by dbyron-sf** [comment]
If cacheSize is 0, that says to me that [allItemsCache](https://github.com/spinnaker/front50/blob/edc56d909d528fe374e9d7120848bf9fc543ceff/front50-core/src/main/java/com/netflix/spinnaker/front50/model/StorageServiceSupport.java#L125) is empty.  (After enabling debug logging) do you see messages like:
```
      log.debug("Running optimized cache refresh");
 ```
or
```
     log.debug("Running unoptimized cache refresh");
 ```
Seems like [fetchAllItemsOptimized](https://github.com/spinnaker/front50/blob/edc56d909d528fe374e9d7120848bf9fc543ceff/front50-core/src/main/java/com/netflix/spinnaker/front50/model/StorageServiceSupport.java#L368) or [fetchAllItems](https://github.com/spinnaker/front50/blob/edc56d909d528fe374e9d7120848bf9fc543ceff/front50-core/src/main/java/com/netflix/spinnaker/front50/model/StorageServiceSupport.java#L371) is returning an empty set.

**Comment by devoxel** [comment]
We see unoptimized cache refresh running (because we don't have versioning on in the bucket).

```
[pool-7-thread-2] c.n.s.f.m.n.DefaultNotificationDAO       : Running unoptimized cache refresh
```

followed by a requests to the last-modified files on the bucket that look something this (simplified the output to only relevant fields).

When we restart front50, we do see requests to the underlying storage to gather data, but it doesn't look for the pipeline/*/specification.json files.

I spotted a suspect diff in the changelog (in https://github.com/spinnaker/front50/pull/1466), see [3ab3bfcb4](https://github.com/spinnaker/front50/commit/3ab3bfcb4648e437ec26a3453de5a18496989d8a)

```
- private static final String DATA_FILENAME = "specification.json";
- private static final String APPLICATION_PERMISSION_DATA_FILENAME = "permission.json";
+ private static final String DATA_FILENAME =
      ObjectType.APPLICATION.getDefaultMetadataFilename(true);
+ private static final String APPLICATION_PERMISSION_DATA_FILENAME =
      ObjectType.APPLICATION_PERMISSION.getDefaultMetadataFilename(true);
```

In this case, a hardcoded DATA_FILENAME is changed to instead actually lookup the filename. 

This seems like it could be correct behaviour, it seems that pipeline-metadata.json is the intended name for this file. But is it possible that prior to this change, all files were stored as specification.json 

I checked filenames contained in the GCS bucket, one in prod which is running 1.34 and one in 1.35. In 1.35, the only files are called specification.json.

So this change seems to have fixed a latent bug that causes an incompatibility in upgrade. 

cc @christosarvanitis who authored the change

```fish
# ( note: this is fish shell, not bash )
# in 1.34 spinnaker instance
for l in *; echo $l && echo (find $l -type f | xargs -n 1 basename | sort | uniq -c) ; end
applications
      1 last-modified     144 permission.json     146 specification.json
delivery
      1 last-modified
notifications
      1 last-modified       2 specification.json
pipeline-strategies
      1 last-modified
pipeline-templates
      1 last-modified       1 specification.json
pipelines
      1 last-modified    1559 specification.json
pluginInfo
      1 last-modified
pluginVersions
      1 last-modified
projects
      1 last-modified       2 specification.json
serviceAccounts
      1 last-modified    1567 specification.json
snapshots
      1 last-modified
tags
      1 last-modified
# in 1.35 spinnaker instance
# note that the old specification.json files that are here are from the 1.34 instance, all pipelines created in 1.35 have the correct name.
for l in *; echo $l && echo (find $l -type f | xargs -n 1 basename | sort | uniq -c) ; end
applications
      1 last-modified     165 permission.json     165 specification.json
delivery
      1 last-modified
notifications
      1 last-modified
pipeline-strategies
      1 last-modified
pipeline-templates
      1 last-modified
pipelines
      1 last-modified     296 pipeline-metadata.json     397 specification.json
pluginInfo
      1 last-modified
pluginVersions
      1 last-modified
projects
      1 last-modified
serviceAccounts
      1 last-modified     296 serviceAccount-metadata.json     362 specification.json
snapshots
      1 last-modified
tags
      1 last-modified
```

**Comment by christosarvanitis** [comment]
Yeah sorry my PR caused this. I have already raised the PR here to fix it https://github.com/spinnaker/front50/pull/1493

Basically my change was not intended to break any compatibility with previous releases but based on how the GCS code looked like i never realised that the ObjectType enum was not used for all the other ObjectTypes. It was hardcoded to be `specification.json` for everything except ApplicationPermissionDAO.

Sorry for the mess

**Comment by shlomodaari** [comment]
You can use the image from here as hotfix for now: https://console.cloud.google.com/artifacts/docker/spinnaker-community/us/docker/front50
We verified that the fix is working!

**Comment by dbyron-sf** [comment]
[Spinnaker 1.35.3](https://spinnaker.io/changelogs/1.35.3-changelog/) contains the fix.

**Comment by christosarvanitis** [comment]
Closing the issue as it has been resolved and released in [1.35.3](https://spinnaker.io/changelogs/1.35.3-changelog/) && [1.34.5](https://spinnaker.io/changelogs/1.34.5-changelog/)
